packages/blocks/src/api/registration.js

```diff
@@ -436,3 +458,23 @@ export const registerBlockStyle = ( blockName, styleVariation ) => {
 export const unregisterBlockStyle = ( blockName, styleVariationName ) => {
 	dispatch( 'core/blocks' ).removeBlockStyles( blockName, styleVariationName );
 };
+
+/**
+ * Registers a new block pattern for the given block.
+ *
+ * @param {string}         blockName Name of the block (example: “core/columns”).
+ * @param {WPBlockPattern} pattern   Object describing a block pattern.
+ */
+export const __experimentalRegisterBlockPattern = ( blockName, pattern ) => {
```

Comment:
@youknowriad - do we even need those APIs in the first place?  I followed `registerBlockStyle` and `unregisterBlockStyle` but we could also use data API directly. What do you think?

Reply:
> @youknowriad - do we even need those APIs in the first place? I followed `registerBlockStyle` and `unregisterBlockStyle` but we could also use data API directly. What do you think?

In my opinion, I think we should have them:

- It's slightly more convenient / obvious to use, than via the equivalent data actions
- Consistency for consistency's sake

Reply:
I used this `wp.blocks.*` API in the examples presenting how you can tweak the Columns block with custom patterns so maybe it's indeed more convenient. The benefit is also that you don't have to worry whether the `core/blocks` store is initialized as it's embedded in the module. Wheres using the data module directly is prone to error in that regard.